src/main/java/de/tum/in/test/api/ast/type/ConditionalType.java

```diff
@@ -0,0 +1,36 @@
+package de.tum.in.test.api.ast.type;
+
+import java.util.Map;
+
+import org.apiguardian.api.API;
+import org.apiguardian.api.API.Status;
+
+import com.github.javaparser.ast.Node;
+import com.github.javaparser.ast.stmt.*;
+
+@API(status = Status.MAINTAINED)
+public enum ConditionalType implements Type {
+	/**
+	 * All conditional types
+	 */
+	ANY(Map.of("If-Statement", IfStmt.class, "Switch-Statement", SwitchStmt.class)), //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
[Implemented part of the third feedback iteration](https://github.com/ls1intum/Ares/pull/248/commits/625a5f2b6d342cb4cbe499e7ea8e9d197d7c5c96): Tenary operators and switch expressions are now covered.

Regarding `if (x) y;` is equivalent to `while (x) { y; break; }`: This is correct, but the AST-Assertions should (in my oppinion) be used to prevent students from using certain expressions and statements, not to prevent them from using certain programming concepts (like branching). In case you would like to prevent the latter, you need to prevent the usage of any expressions or statement, which could provide some form of branching functionality. This might be, aside from the already mentioned while-loop also try-catch statements (branching based on whether an exception is thrown or not). But this is up to the test case creator.